File: 229-majority-element-ii/majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        unordered_map<int, int> countMap;
        vector<int> result;
        int n = nums.size() / 3;

        for (int num : nums) countMap[num]++;
        for (auto& [num, count] : countMap) if (count > n) result.push_back(num);
        
        return result;
    }
};

File: 229-majority-element-ii/majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        Map<Integer, Integer> countMap = new HashMap<>();
        List<Integer> result = new ArrayList<>();
        int n = nums.length/3;
        for (int num : nums) countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        for (Map.Entry<Integer, Integer> entry : countMap.entrySet()) {
            if (entry.getValue() > n) result.add(entry.getKey());
        }
        return result;
    }
}